## Importing dependies
- Then checking various things in the dataset
- Doing the train_test_split
- Using linear regression to the the model score on the specific dataset

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('https://github.com/ybifoundation/Dataset/raw/main/Admission%20Chance.csv')

In [3]:
df

Serial No  GRE Score  TOEFL Score  University Rating   SOP  LOR   CGPA  \
0            1        337          118                  4   4.5   4.5  9.65   
1            2        324          107                  4   4.0   4.5  8.87   
2            3        316          104                  3   3.0   3.5  8.00   
3            4        322          110                  3   3.5   2.5  8.67   
4            5        314          103                  2   2.0   3.0  8.21   
..         ...        ...          ...                ...   ...   ...   ...   
395        396        324          110                  3   3.5   3.5  9.04   
396        397        325          107                  3   3.0   3.5  9.11   
397        398        330          116                  4   5.0   4.5  9.45   
398        399        312          103                  3   3.5   4.0  8.78   
399        400        333          117                  4   5.0   4.0  9.66   

     Research  Chance of Admit   
0           1              0.92  
1           1              0.76  
2           1              0.72  
3           1              0.80  
4           0              0.65  
..        ...               ...  
395         1              0.82  
396         1              0.84  
397         1              0.91  
398         0              0.67  
399         1              0.95  

[400 rows x 9 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Serial No          400 non-null    int64  
 1   GRE Score          400 non-null    int64  
 2   TOEFL Score        400 non-null    int64  
 3   University Rating  400 non-null    int64  
 4    SOP               400 non-null    float64
 5   LOR                400 non-null    float64
 6   CGPA               400 non-null    float64
 7   Research           400 non-null    int64  
 8   Chance of Admit    400 non-null    float64
dtypes: float64(4), int64(5)
memory usage: 28.2 KB


In [6]:
df.isnull().sum()

Serial No            0
GRE Score            0
TOEFL Score          0
University Rating    0
 SOP                 0
LOR                  0
CGPA                 0
Research             0
Chance of Admit      0
dtype: int64

In [8]:
df.describe()

Serial No   GRE Score  TOEFL Score  University Rating         SOP  \
count  400.000000  400.000000   400.000000         400.000000  400.000000   
mean   200.500000  316.807500   107.410000           3.087500    3.400000   
std    115.614301   11.473646     6.069514           1.143728    1.006869   
min      1.000000  290.000000    92.000000           1.000000    1.000000   
25%    100.750000  308.000000   103.000000           2.000000    2.500000   
50%    200.500000  317.000000   107.000000           3.000000    3.500000   
75%    300.250000  325.000000   112.000000           4.000000    4.000000   
max    400.000000  340.000000   120.000000           5.000000    5.000000   

             LOR         CGPA    Research  Chance of Admit   
count  400.000000  400.000000  400.000000        400.000000  
mean     3.452500    8.598925    0.547500          0.724350  
std      0.898478    0.596317    0.498362          0.142609  
min      1.000000    6.800000    0.000000          0.340000  
25%      3.000000    8.170000    0.000000          0.640000  
50%      3.500000    8.610000    1.000000          0.730000  
75%      4.000000    9.062500    1.000000          0.830000  
max      5.000000    9.920000    1.000000          0.970000

In [9]:
df.columns

Index(['Serial No', 'GRE Score', 'TOEFL Score', 'University Rating', ' SOP',
       'LOR ', 'CGPA', 'Research', 'Chance of Admit '],
      dtype='object')

In [12]:
y = df['Chance of Admit ']

In [14]:
X = df.drop(['Serial No','Chance of Admit '],axis=1)

In [17]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,train_size = 0.2,random_state=2529)

In [18]:
X_train.shape, X_test.shape,y_train.shape,y_test.shape

((80, 7), (320, 7), (80,), (320,))

In [21]:
from sklearn.linear_model import LinearRegression
model = LinearRegression()

In [22]:
model.fit(X_train,y_train)

LinearRegression()

In [24]:
model.intercept_

-1.174879623744979

In [29]:
model.coef_

array([ 0.00226941, -0.00185351,  0.00083679, -0.00136259,  0.02192977,
        0.15262218,  0.00898032])

In [30]:
y_pred = model.predict(X_test)

In [31]:
y_pred

array([0.73181977, 0.73683927, 0.71987944, 0.70621253, 0.59532402,
       0.93156035, 0.9460236 , 0.7586047 , 0.81109468, 0.64897416,
       0.61352985, 0.80472068, 0.54475149, 0.81674792, 0.83425414,
       0.68198618, 0.66242457, 0.71955285, 0.75992689, 0.87196616,
       0.65462163, 0.85462379, 0.85965541, 0.94887923, 0.72507473,
       0.64539534, 0.68451098, 0.59924644, 0.57340612, 0.56669665,
       0.55294789, 0.75715149, 0.74365776, 0.79675505, 0.61260193,
       0.57551386, 0.46306905, 0.85288628, 0.88541511, 0.80589867,
       0.76475554, 0.73027224, 0.68755934, 0.87000038, 0.75198792,
       0.78730971, 0.98176019, 0.87765635, 0.62938903, 0.52197749,
       0.61637255, 0.64803327, 0.83491467, 0.47109308, 0.69895437,
       0.8076501 , 0.70407328, 0.65705394, 0.86756785, 0.66770345,
       0.65713335, 0.90774113, 0.63915117, 0.68551381, 0.75358083,
       0.68995295, 0.62091408, 0.94008722, 0.56981656, 0.90888222,
       0.61855963, 0.91510852, 0.61316979, 0.72876412, 0.56215

In [32]:
from sklearn.metrics import mean_absolute_error, mean_absolute_percentage_error, mean_squared_error, r2_score

In [34]:
mean_absolute_error(y_test,y_pred)

0.04888864807409543

In [35]:
mean_absolute_percentage_error(y_test,y_pred)

0.0814022821302939

In [36]:
mean_squared_error(y_test,y_pred)

0.004848556366255484

In [37]:
r2_score(y_test,y_pred)

0.765873950722899

### Thus, the accuracy of the model is 76% using the Linear Regression.